src/runner/at-driver.js

```diff
@@ -128,6 +141,18 @@ export class ATDriver {
       }
     }
   }
+
+  /**
+   *
+   * @param {[{name: string, value: string | boolean}]} settings
+   */
+  async setSettings(settings) {
+    if (settings?.length) {
```

Comment:
Is there any reason to forgive invalid invocations of this method? None of the call sites in this patch appear to be susceptible, and I have to assume that any future code would be erroneous and best reported with an exception.

Reply:
Probably not? I still have the tendency to program defensively though, want I just remove the if?